js/optional-chaining-is-amazing-heres-why/optional-chaining-is-amazing-here-is-why.md

```diff
@@ -0,0 +1,154 @@
+**Optional Chaining** is the new javascript operator, which is included as part of EcmaScript 2020. The operator works for objects, arrays, and functions. They don't enable anything in terms of functionality, but they make the code a lot easy to read and write. Let's see how:
+
+I'm sure that we all in our coding experience have faced a common problem, for which we already have a solution.
+
+**The Problem**
+
+```javascript
+{
+    data: {
+        user: {
+            fullname: "Soumya Mishra"
+            wishlist: ["wId_101", "wId_102"]
+        }
+    }
+}
+
+// Error prone-version, could throw.
+const wishlistItem = db.user.wishlist.length;
+```
+
+Let's assume that the above object is for loggedIn user details along with users wishlist data. Now, there can be users who have not added anything to their wishlist, so the wishlist property in the object might not be present for those users. But we have to pull the wishlist Ids from this object and make another request to fetch data.
+
+Till date, we choose any 1 from the below ways to handle such scenarios.
+
+**The Solution**
+
+```javascript
+const wishlistIds =
+    data ?
+    (data.user ?
+        (data.user.wishlist ?
+            data.user.wishlist :
+            []) :
+        []) :
+    [];
+const wishlistItemCount = wishlistIds.length;
+
+OR
+
+const wishlistIds = [];
+if (data && data.user && data.user.wishlist) {
+    wishlistIds = data.user.wishlist;
+}
+const wishlistItemCount = wishlistIds.length;
+```
+
+This is something just for the small object, assume it for more complex object structure. Even if one writes the code correctly, it will not provide any readability.
+
+**The Magic Solution**
+
+Now we will see how we can achieve the above solution in just 1 line.
+
+```javascript
+const wishlistItemCount = data?.user?.wishlist?.length;
+```
+
+This magic solution has been achieved through **Optional Chaining (?.) Operator**.
+
+Now let's see how this Operator works with different data types in javascript. It only works with **Arrays**, **Objects** & **Functions**.
```

Comment:
Just curious, where does it not work?

Reply:
I corrected the sentence plus the Into to the topic as it works basically to validate the properties of an Object. The type of those properties can be of anything. The earlier sentence was confusing. Thanks

Reply:
👍